### ISSUE TITLE
S3459 FP with private inner class fields initialized in the outer class

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#4757]
### Description

Found [on sonarcloud](https://sonarcloud.io/project/issues?id=sonaranalyzer-dotnet&issues=AXsNH8AULrVS-fD458_u&open=AXsNH8AULrVS-fD458_u&pullRequest=4756).

It seems to repro only on generic classes. Only if `X` is used inside one of the methods - if it is not used, it is not reported.

### Repro steps

```
public sealed class OuterFoo
{
    private static class InnerFactory
    {
        public static InnerFoo<string> Instance = new InnerFoo<string> { X = "X" };
    }

    private class InnerFoo<T>
    {
        public string X; // FP S3459 - remove unassigned field `X` or set its value

        public void Foo()
        {
            Console.Write(X);
        }
    }
}

```

### Expected behavior

It is a private class initialized in another private class where the field is used - the issue should not be reported.

### Actual behavior

Reports a false positive.

### Known workarounds

None.

### Related information

It reproes on SonarCloud and in SonarLint for Visual Studio.

* C#/VB.NET Plugins version 8.26 
* Visual Studio version 16.10.3
* MSBuild / dotnet version 16.9 (on AZP)
* SonarScanner for .NET version (if used) 5.2.2
